static/app/components/nav/utils.tsx

```diff
@@ -0,0 +1,165 @@
+import {type HTMLProps, useMemo} from 'react';
+import type {LocationDescriptor} from 'history';
+
+import {isItemActive} from 'sentry/components/sidebar/sidebarItem';
+import {SIDEBAR_NAVIGATION_SOURCE} from 'sentry/components/sidebar/utils';
+import type {FeatureDisabledHooks, WithoutPrefix} from 'sentry/types/hooks';
+import normalizeUrl from 'sentry/utils/url/normalizeUrl';
+import type {useLocation} from 'sentry/utils/useLocation';
+
+export const NAV_DIVIDER = Symbol('divider');
+
+interface FeatureCheck {
+  features: string | string[];
+  hook?: WithoutPrefix<keyof FeatureDisabledHooks>;
+}
+
+interface NavItem {
+  label: string;
+  check?: FeatureCheck;
+}
+
+export interface SidebarItem extends SidebarItemBase {
+  to: string;
+  submenu?: SubmenuItem[];
+}
+
+export interface SidebarItemBase extends NavItem {
+  icon: JSX.Element;
+}
+interface SidebarItemTopLevel extends SidebarItemBase {
+  to: string;
+  submenu?: never;
+}
+
+interface SidebarItemWithChildren extends SidebarItemBase {
+  submenu: (SubmenuItem | false)[];
+  to?: never;
+}
+
+export interface SubmenuItem extends NavItem {
+  to: string;
+}
+
+export type NavItemRaw =
+  | SidebarItemTopLevel
+  | SidebarItemWithChildren
+  | typeof NAV_DIVIDER
+  | undefined
+  | false;
+
+export function resolveSidebarItem(
+  item: SidebarItemTopLevel | SidebarItemWithChildren
+): SidebarItem {
+  if (item.submenu) {
+    const submenu = item.submenu.filter(subitem => !!subitem) as SubmenuItem[];
+    const [{to}] = submenu;
+    return {...item, submenu, to};
+  }
+  return item;
+}
```

Comment:
Why do we need this?

Reply:
Removed in [92d70b5](https://github.com/getsentry/sentry/pull/77665/commits/92d70b59ab5a518fa704b5483e958590d2fbe50c).